### ISSUE TITLE
FP Fly camera needed

### ISSUE THREAD

**Issue by RepairUnit3k6** [simlu/voxelshop#294]
We need the first person fly camera mode. Orbit camera is good but we need the first person fly camera to effectively work with small details and interior models (buildings).We need camera like in editor of GoDot Engine. W,A,S & D for vertical movement, Q&E for horizontal movement and active mouse look. Something like build mode in Robocraft. I don't know java but it should't be very hard (Or at least it wouldn't be very hard in C#)

**Comment by simlu** [comment]
Agreed. It shouldn't be very hard. The camera is abstracted already as far 
as I recall.

Feel free to take a look and ask for some feedback on how to implement this 
and do a pull request (voxelshop is open source). I maintain too many 
projects at this point to get to this change in, in a reasonable timeframe 
unfortunately. Trust me, I wish my day had more hours :)

Somewhat related: I'm assuming you know that you can change the rotation 
center in view mode?

**Comment by Garrom** [comment]
> Agreed. It shouldn't be very hard. The camera is abstracted already as far as I recall. Feel free to take a look and ask for some feedback on how to implement this and do a pull request (voxelshop is open source). I maintain too many projects at this point to get to this change in, in a reasonable timeframe unfortunately. Trust me, I wish my day had more hours :) Somewhat related: I'm assuming you know that you can change the rotation center in view mode?

Yeah I know that I can move center of rotation around, pretty much only way how to actually move around.I checked [this file](voxelshop/src/main/java/com/vitco/app/core/CCamera.java), I guess it is camera controller. It looks somewhat similar to C# but I don't understand it much. I understand your struggle with time, 58 repos is not some small number. I'd love to write it and push but I can't understand Java. At start, I was doing GML then C#(Unity3D) for some time and now GDScript(GoDot). Last time I touched java was Minecraft troubleshooting.
